[17/Dec/2013:19:58:20] ENGINE Bus STARTING
[17/Dec/2013:19:58:20] ENGINE Started monitor thread '_TimeoutMonitor'.
[17/Dec/2013:19:58:21] ENGINE Serving on 0.0.0.0:8080
[17/Dec/2013:19:58:22] ENGINE Bus STARTED
[17/Dec/2013:19:58:24] ENGINE Bus STOPPING
[17/Dec/2013:19:58:25] ENGINE HTTP Server cherrypy._cpwsgi_server.CPWSGIServer(('0.0.0.0', 8080)) shut down
[17/Dec/2013:19:58:25] ENGINE Stopped thread '_TimeoutMonitor'.
[17/Dec/2013:19:58:25] ENGINE Bus STOPPED
[17/Dec/2013:19:58:25] ENGINE Bus EXITING
[17/Dec/2013:19:58:25] ENGINE Bus EXITED
[17/Dec/2013:19:58:25] ENGINE Waiting for child threads to terminate...
[17/Dec/2013:19:58:25] ENGINE Waiting for thread MainThread.
[17/Dec/2013:20:07:16] ENGINE Bus STARTING
[17/Dec/2013:20:07:16] ENGINE Bus STARTING
[17/Dec/2013:20:07:16] ENGINE Started monitor thread '_TimeoutMonitor'.
[17/Dec/2013:20:07:16] ENGINE Started monitor thread '_TimeoutMonitor'.
[17/Dec/2013:20:07:16] ENGINE Serving on 0.0.0.0:8080
[17/Dec/2013:20:07:16] ENGINE Serving on 0.0.0.0:8080
[17/Dec/2013:20:07:17] ENGINE Bus STARTED
[17/Dec/2013:20:07:17] ENGINE Bus STARTED
[17/Dec/2013:20:07:33] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:07:33] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:07:42] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:07:42] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:07:44] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:07:44] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:07:48] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:07:48] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:08:32] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:08:32] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:08:35] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:08:35] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:08:38] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:08:38] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:08:47] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:08:47] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:09:01] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:09:01] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:09:05] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:09:05] HTTP Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/_cprequest.py", line 656, in respond
    response.body = self.handler()
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/encoding.py", line 188, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpdispatch.py", line 34, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 54, in do
    ans = func(*args, **kwargs)
  File "/usr/lib/calibre/calibre/library/server/utils.py", line 108, in authenticate
    self.secret)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 342, in digest_auth
    ha1 = get_ha1(realm, auth.username)
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 55, in get_ha1
    return md5_hex('%s:%s:%s' % (username, realm, password))
  File "/usr/lib/python2.7/site-packages/cherrypy/lib/auth_digest.py", line 30, in <lambda>
    md5_hex = lambda s: md5(ntob(s)).hexdigest()
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 58, in ntob
    assert_native(n)
  File "/usr/lib/python2.7/site-packages/cherrypy/_cpcompat.py", line 96, in assert_native
    raise TypeError("n must be a native str (got %s)" % type(n).__name__)
TypeError: n must be a native str (got unicode)

[17/Dec/2013:20:12:36] ENGINE Bus STOPPING
[17/Dec/2013:20:12:36] ENGINE Bus STOPPING
[17/Dec/2013:20:12:37] ENGINE HTTP Server cherrypy._cpwsgi_server.CPWSGIServer(('0.0.0.0', 8080)) shut down
[17/Dec/2013:20:12:37] ENGINE HTTP Server cherrypy._cpwsgi_server.CPWSGIServer(('0.0.0.0', 8080)) shut down
[17/Dec/2013:20:12:37] ENGINE Stopped thread '_TimeoutMonitor'.
[17/Dec/2013:20:12:37] ENGINE Stopped thread '_TimeoutMonitor'.
[17/Dec/2013:20:12:37] ENGINE Bus STOPPED
[17/Dec/2013:20:12:37] ENGINE Bus STOPPED
[17/Dec/2013:20:12:37] ENGINE Bus EXITING
[17/Dec/2013:20:12:37] ENGINE Bus EXITING
[17/Dec/2013:20:12:37] ENGINE Bus EXITED
[17/Dec/2013:20:12:37] ENGINE Bus EXITED
[17/Dec/2013:20:12:37] ENGINE Waiting for child threads to terminate...
[17/Dec/2013:20:12:37] ENGINE Waiting for child threads to terminate...
[17/Dec/2013:20:12:37] ENGINE Waiting for thread MainThread.
[17/Dec/2013:20:12:37] ENGINE Waiting for thread MainThread.
[18/Dec/2013:17:44:44] ENGINE Bus STARTING
[18/Dec/2013:17:44:44] ENGINE Bus STARTING
[18/Dec/2013:17:44:44] ENGINE Bus STARTING
[18/Dec/2013:17:44:44] ENGINE Started monitor thread '_TimeoutMonitor'.
[18/Dec/2013:17:44:44] ENGINE Started monitor thread '_TimeoutMonitor'.
[18/Dec/2013:17:44:44] ENGINE Started monitor thread '_TimeoutMonitor'.
[18/Dec/2013:17:44:44] ENGINE Serving on 0.0.0.0:8080
[18/Dec/2013:17:44:44] ENGINE Serving on 0.0.0.0:8080
[18/Dec/2013:17:44:44] ENGINE Serving on 0.0.0.0:8080
[18/Dec/2013:17:44:46] ENGINE Bus STARTED
[18/Dec/2013:17:44:46] ENGINE Bus STARTED
[18/Dec/2013:17:44:46] ENGINE Bus STARTED
[13/Mar/2014:23:12:39] ENGINE Bus STARTING
[13/Mar/2014:23:12:39] ENGINE Started monitor thread '_TimeoutMonitor'.
[13/Mar/2014:23:12:39] ENGINE Serving on 0.0.0.0:8080
[13/Mar/2014:23:12:41] ENGINE Bus STARTED
[13/Mar/2014:23:29:48] ENGINE Bus STOPPING
[13/Mar/2014:23:29:49] ENGINE HTTP Server cherrypy._cpwsgi_server.CPWSGIServer(('0.0.0.0', 8080)) shut down
[13/Mar/2014:23:29:49] ENGINE Stopped thread '_TimeoutMonitor'.
[13/Mar/2014:23:29:49] ENGINE Bus STOPPED
[13/Mar/2014:23:29:49] ENGINE Bus EXITING
[13/Mar/2014:23:29:49] ENGINE Bus EXITED
[13/Mar/2014:23:29:49] ENGINE Waiting for child threads to terminate...
[13/Mar/2014:23:29:49] ENGINE Waiting for thread MainThread.
[26/Sep/2014:12:43:53] ENGINE Bus STARTING
[26/Sep/2014:12:43:53] ENGINE Started monitor thread '_TimeoutMonitor'.
[26/Sep/2014:12:43:53] ENGINE Serving on http://0.0.0.0:8080
[26/Sep/2014:12:43:55] ENGINE Bus STARTED
[26/Sep/2014:15:16:27] ENGINE Bus STOPPING
[26/Sep/2014:15:16:28] ENGINE HTTP Server cherrypy._cpwsgi_server.CPWSGIServer(('0.0.0.0', 8080)) shut down
[26/Sep/2014:15:16:28] ENGINE Stopped thread '_TimeoutMonitor'.
[26/Sep/2014:15:16:28] ENGINE Bus STOPPED
[26/Sep/2014:15:16:28] ENGINE Bus EXITING
[26/Sep/2014:15:16:28] ENGINE Bus EXITED
[26/Sep/2014:15:16:32] ENGINE Waiting for child threads to terminate...
